examples/quilkin-filter-example/Cargo.toml

```diff
@@ -22,13 +22,17 @@ repository = "https://github.com/googleforgames/quilkin"
 edition = "2018"
 
 [dependencies]
-quilkin = "0.1.0"
-tokio = { version = "1", features = ["full"]}
-prost = "0.7"
-prost-types = "0.7"
+# If lifting this example, you will want to be explicit abou the Quilkin version, e.g.
+# quilkin = "0.2.0"
+quilkin = { path = "../../" }
+tokio = { version = "1", features = ["full"] }
+prost = "0.8"
+prost-types = "0.8"
 serde = "1.0"
 serde_yaml = "0.8"
 bytes = "1.0"
 
 [build-dependencies]
-prost-build = "0.7"
+prost-build = "0.8"
+
+[workspace]
```

Comment:
Had to add this `[workspace]` field here, otherwise I would get the error:

`Error during execution of `cargo metadata`: error: current package believes it's in a workspace when it's not:` -- I'm not sure I really get cargo workspaces tbh, so if you have any feedback here, I would appreciate it.

Reply:
All cargo projects (even ones with one crate) are workspaces, starting from the `Cargo.toml` and recursively throughout the directory. So if you have a project with multiple crates, cargo needs to figure out if it's part of the parent workspace, or if it's in a new workspace (this will determine what crates to build, where the target directory is, etc). One way to do that is to add `[workspace]` in the `Cargo.toml`, which is explicitly saying "This is the start of a new workspace". The other option is using the `members` and `exclude` fields in the parent `[workspace]`. I would probably go with either of these.

- `members = ["examples/quilkin-filter-example", ...]`, add it to the `members` field, which means it will be built as part of `cargo build --workspace`.
- `exclude = ["examples/quilkin-filter-example", ...]`, add it to the `exclude` field, which means it will **not** be built as part of `cargo build --workspace`, and you need to build it explicitly yourself.

Reply:
Thanks for the detailed explanation - that helped a lot!

I'm definitely leaning towards having it be seperate, and that way if you are developing against quilkin you don't _have_ to process the example of every build and test.

I just tried using `exclude` in `quilkin/Cargo.toml`, and I'm still getting the error:

```
Error during execution of `cargo metadata`: error: current package believes it's in a workspace when it's not:
current:   /home/markmandel/workspace/quilkin/examples/quilkin-filter-example/Cargo.toml
workspace: /home/markmandel/workspace/quilkin/Cargo.toml

this may be fixable by adding `examples/quilkin-filter-example` to the `workspace.members` array of the manifest located at: /home/markmandel/workspace/quilkin/Cargo.toml
Alternatively, to keep it out of the workspace, add the package to the `workspace.exclude` array, or add an empty `[workspace]` table to the package's manifest.
```

Seems like leaving `[workspace]` in the example `Cargo.toml` is the only way.